llvm/lib/TableGen/TGLexer.cpp

```diff
@@ -42,6 +42,25 @@ struct {
   { tgtok::Endif, "endif" },
   { tgtok::Define, "define" }
 };
+
+// Returns true if `MacroName` is a valid macro name. Valid macro names match
+// the regular expression [a-zA-Z_][0-9a-zA-Z_]* (see prepLexMacroName).
```

Comment:
Does prepLexMacroName need a backreference to this function, so they can both be kept in sync?

Reply:
I changed the code to have this function usable in both places to we have a single copy. That means most of the rest of the comments do not apply anymore. Can you please take a fresh look?

Reply:
FWIW it looked fine to me with a bit of code duplication.